spec/models/user_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { described_class.new(name: 'John') }
+
+  describe 'associations' do
+    it { should have_many(:posts).with_foreign_key(:author_id).dependent(:destroy) }
+    it { should have_many(:comments).with_foreign_key(:author_id) }
+    it { should have_many(:likes).with_foreign_key(:author_id).class_name('Like').dependent(:destroy) }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_numericality_of(:posts_count).only_integer.is_greater_than_or_equal_to(0) }
+  end
+
+  describe '#last_three_posts' do
+    let!(:user) { create(:user) }
```

Comment:
- **[OPTIONAL]** This test returns an error, please fix it to pass them all.

![image](https://github.com/Zeph180/blog-it/assets/58642949/ff443325-627e-43dc-879c-9a897f54efef)